### ISSUE TITLE
Is there i18n support?

### ISSUE THREAD

**Issue by xiao-cao-x** [microsoft/Xaml-Controls-Gallery#787]
My English is very poor, can `Xaml-Controls-Gallery` support i18n like [Flutter Gallery](https://github.com/flutter/gallery/)?

**Comment by chingucoding** [comment]
As of right now, there is no i18n support yet, however this is something that could be added. @stmoy thoughts maybe?

**Comment by stmoy** [comment]
This is a very interesting idea! I suspect this would have two parts:
1. Add the architecture to support i18n 
2. Over time, add translations

Although I think this is generally a good idea to explore, I'm honestly not too familiar with how difficult it would be to add. I'd imagine it would require moving all the text to resource files and loading them in dynamically? 

I also wonder if the mechanisms required to do this would be different between the WinUI2/UWP version and the WinUI3 version?

TL;DR - I think this is a good idea, I just don't have enough knowledge to know what it entails. 

CC: @gregwoo-microsoft

**Comment by chingucoding** [comment]
@stmoy You are correct in assuming that this would require moving the text into resource files. Since we would want to probably localize everything, this would mean, having to touch every sample page and move the text into resource files. Regarding UWP&WinUI 2 vs WinUI 3, those should work the same way so not much work would be needed to port it to one version to the other version.

**Comment by mohdabsharmohamadsiddik** [comment]
> > > # - [ ] - @VSC-Service-Account ~~````~~@opbld37 @https://absharsiddik170979abshar.jetbrains.space/on boardinghttps://www.bing.com/search?q=https%3a%2f%2fabsharsiddik170979abshar.jetbrains.space%2fonboarding&setmkt=en-US&PC=EMMX40&scope=web&first=39&FORM=PERE2
DESIGNER'S INC DRESSER OVERALL'S LTD dashboard.stripe.com

**Comment by gregwoo-microsoft** [comment]
Just want to give you a quick update: this idea is in the backlog but is not a top priority right now. Rest assured that we will do our best to implement it in the future. Will update this thread if there is progress.

**Comment by KaranocaVe** [comment]
How's everything going?

**Comment by marcelwgn** [comment]
I'm afraid the priority has not changed.

**Comment by Jay-o-Way** [comment]
https://github.com/AndrewKeepCoding/WinUI3Localizer
Would be nice to try

**Comment by Jay-o-Way** [comment]
> Over time, add translations

Gotta decide on the translation platform. Internal hidden Microsoft team or open like Crowdin?

**Comment by ghost1372** [comment]
> https://github.com/AndrewKeepCoding/WinUI3Localizer
> Would be nice to try

Why?
When winui itself support localization for packged and unpackaged

**Comment by AndrewKeepCoding** [comment]
> https://github.com/AndrewKeepCoding/WinUI3Localizer
Would be nice to try

While it would be nice for me (the maintainer), but I think the WinUI Gallery should stick to WinUI built-in features whenever possible.🙂
Unless you need dynamic localization of course.🔥

**Comment by Jay-o-Way** [comment]
You're right. I didn't read well enough. Thought it would help with the initial conversion.

**Comment by ghost1372** [comment]
> You're right. I didn't read well enough. Thought it would help with the initial conversion.

v1.6 added support for **Microsoft.Windows.Globalization.ApplicationLanguages** which can be used in Packaged/UnPackaged and by refreshing Frame Content you can get a dynamic localization
https://github.com/microsoft/WindowsAppSDK/pull/4523

**Comment by Jay-o-Way** [comment]
Hooking back on the initial comment - I want to advise the possibility to split resources into different files.

**Comment by KaranocaVe** [comment]
very good
I'm ready to private translations.
